src/main/java/com/google/sps/data/UserInfo.java

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public final class UserInfo {
```

Comment:
In my opinion, `UserInfo` can be expanded to encapsulate more functionality. If you chose to do some of the suggestions here `UserInfo` will probably be better named as `User`.

Some functionality that we could move here that comes to mind:
- A static method that takes a `com.google.appengine.api.datastore.Entity` and returns a new `UserInfo` object with all fields populated by that Entity (i.e. what you did in `convertEntitytoUserInfo`)
- A method that converts a `UserInfo` to `Entity` suitable for writing into Datastore. `src/main/java/com/google/sps/servlets/UserServlet.java` lines 108-114 can just be something like `datastore.put(UserInfo(userId, nickname, createdEvents, bookmarkedEvents, currentChallengeId, challengeStatuses).toEntity())`
- If you do both bullets above, you should interact with the `UserInfo` object all the time, only using `Entity` when loading/storing from Datastore. `src/main/java/com/google/sps/servlets/UserServlet.java` lines 135-147 can therefore be modifying a `UserInfo` object directly, which can help improve clarity as well as reduce the brittleness of the code. Modifying Entity properties directly (in `updateCurrentChallenge` and `updateChallengeStatus`) can be brittle because setProperty is untyped. By always interacting with a typed object (`UserInfo`) you leverage the compiler to catch potential bugs.

Reply:
Added the functionality you suggested in the bullet points -- Named it UserInfo instead of User because there is also a User class from the Google API that I used